controllers/repo_manager/api.go

```diff
@@ -150,6 +150,21 @@ func (r *RepoManagerReconciler) pulpApiController(ctx context.Context, pulp *rep
 		return ctrl.Result{Requeue: requeue}, err
 	}
 
+	// Ensure the secret data is as expected
+	serverSecret := &corev1.Secret{}
+	r.Get(ctx, types.NamespacedName{Name: pulp.Name + "-server", Namespace: pulp.Namespace}, serverSecret)
+	expectedServerSecret := pulpServerSecret(funcResources)
+	if requeue, err := reconcileObject(funcResources, expectedServerSecret, serverSecret, conditionType); err != nil || requeue {
+
+		// when requeue==true it means the secret changed so we need to redeploy api and content pods to get the new settings.py
```

Comment:
maybe a log might be useful here

Reply:
oh... good point!
I'll open an issue for this.